File: src/lessons/welcome/ConditionsEntity.java
package lessons.welcome;

import jlm.universe.bugglequest.SimpleBuggle;

public class ConditionsEntity extends SimpleBuggle {
	@Override
	public void run() { 
		/* BEGIN TEMPLATE */
		/* BEGIN SOLUTION */
		if (isFacingWall())
			backward();
		else
			forward();
		/* END SOLUTION */
		/* END TEMPLATE */
	}
}

File: src/lessons/welcome/conditions/ConditionsEntity.scala
package lessons.welcome.conditions;

import plm.universe.bugglequest.SimpleBuggle;

class ConditionsEntity extends SimpleBuggle {
	override def run() { 
		/* BEGIN SOLUTION */
		if (isFacingWall())
			stepBackward();
		else
			stepForward();
		/* END SOLUTION */
	}
}
